tutorials/build-first-cluster.md

```diff
@@ -0,0 +1,299 @@
+:::
+
+## Prerequisites and dependencies
+
+To successfully complete this tutorial, you will need:
+
+
+* 8 cpus, 20GB RAM, and 40GB storage available
+* [Multipass installed](https://canonical.com/multipass/install)
+* An active wifi connection
+* A local copy of the `charmed-hpc-tutorial-cloud-init.yaml`
+
+## Create Multipass VM
+
+Using the `charmed-hpc-tutorial-cloud-init.yaml`, launch a Multipass VM:
+
+:::{code-block} shell
+multipass launch 24.04 --name charmed-hpc-tutorial-vm --cloud-init charmed-hpc-tutorial-cloud-init.yml --memory 16G --disk 40G --cpus 8
+:::
+
+<!-- Rephrase this section -->
+Note that the virtual machine launch process may take ten minutes or longer to complete. If the instance states that it has failed to launch due to timing out, check `multipass list`{l=shell} to confirm the status of the instance as it may have actually successfully created the vm. If the `State` is `Running`, then the vm was launched successfully and may simply be completing the cloud-init process.
+<!-- Steps if the vm does not say running? -->
+
+The cloud init process creates and configures our lxd machine cloud `localhost` with the `charmed-hpc-controller` juju controller and our `charmed-hpc-k8s` Kubernetes control cloud.
+<!-- Add ref arch pieces -->
+
+To check the status of cloud-init, first, enter the vm:
+
+:::{code-block} shell
+multipass shell charmed-hpc-tutorial-vm
+:::
+
+Then check `cloud init status`{l=shell} with:
+
+:::{code-block} shell
+cloud-init status --long
+:::
+
+Which will show something similar to the following when complete:
+
+:::{terminal}
+:input: cloud-init status --long
+status: done
+extended_status: done
+boot_status_code: enabled-by-genertor
+last_update: Thu, 01 Jan 1970 00:03:45 +0000
+detail: DataSourceNoCloud [seed=/dev/sr0]
+errors: []
+recoverable_errors: {}
+:::
+
+<!-- Quick commands to test that various cloud init pieces have been set up correctly:
+
+`juju status -m controller` should show...
+`juju clouds` should show...
+ -->
+
+## Deploy Slurm and file system
+
+Next, we will deploy Slurm as the resource management and job scheduling service.
+
+First create the `slurm` model that will hold the
+deployment in our cloud `localhost`:
+
+:::{code-block} shell
+juju add-model slurm localhost
+:::
+
+Then deploy the Slurm management daemon `slurmctld`:
+
+:::{code-block} shell
+juju deploy slurmctld --base "ubuntu@24.04" --channel "edge" --constraints="virt-type=virtual-machine"
+:::
+
+and the Slurm compute node daemon with partition name 'tutorial-partition' and two nodes:
+
+:::{code-block} shell
+juju deploy slurmd tutorial-partition --base "ubuntu@24.04" --channel "edge" --constraints="virt-type=virtual-machine" -n 2
+:::
+
+and the authentication and credential kiosk daemon `sackd`:
+
+:::{code-block} shell
+juju deploy sackd --base "ubuntu@24.04" --channel "edge" --constraints="virt-type=virtual-machine"
+:::
```

Comment:
IMO we should group the deployment of the slurm charms together in a single block, explaining beforehand what are the components. It reduces visual clutter, since you don't need to alternate between reading text and reading shell commands.

Reply:
Maybe this could also apply to other commands in the tutorial as a whole.